include/irr/asset/format/encodePixels.h

```diff
@@ -499,23 +499,23 @@ namespace irr { namespace video
     {
         int8_t* pix = reinterpret_cast<int8_t*>(_pix);
         for (uint32_t i = 0u; i < 3u; ++i)
-            pix[i] = static_cast<int8_t>(_input[i] * 127.);
+            pix[i] = static_cast<int8_t>(core::clamp(_input[i] * 127., -127 + 0.4, 127.4));
     }
 	
     template<>
     inline void encodePixels<asset::EF_R8G8B8_USCALED, double>(void* _pix, const double* _input)
     {
         uint8_t* pix = reinterpret_cast<uint8_t*>(_pix);
         for (uint32_t i = 0u; i < 3u; ++i)
-            pix[i] = static_cast<int8_t>(_input[i]);
+            pix[i] = static_cast<int8_t>(core::clamp(_input[i], 0.4, 255.4));
     }
 	
     template<>
     inline void encodePixels<asset::EF_R8G8B8_SSCALED, double>(void* _pix, const double* _input)
     {
         int8_t* pix = reinterpret_cast<int8_t*>(_pix);
         for (uint32_t i = 0u; i < 3u; ++i)
-            pix[i] = static_cast<int8_t>(_input[i]);
+            pix[i] = static_cast<int8_t>(core::clamp(_input[i], 0.4, 255.4));
```

Comment:
wrong range... maybe make a template intermediate function calling `core::clamp` that will deduce the limits from integer type?